File: joiner-test-support/src/main/java/cz/encircled/joiner/test/model/Group.java
package cz.encircled.joiner.test.model;

import javax.persistence.*;
import java.util.Set;

/**
 * @author Kisel on 21.01.2016.
 */
@Entity
@Table(name = "test_group")
public class Group extends AbstractEntity {

    @ManyToMany(fetch = FetchType.LAZY, mappedBy = "groups")
    private Set<User> users;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "group")
    private Set<Status> statuses;

    public Set<User> getUsers() {
        return users;
    }

    public void setUsers(Set<User> users) {
        this.users = users;
    }

    public Set<Status> getStatuses() {
        return statuses;
    }

    public void setStatuses(final Set<Status> statuses) {
        this.statuses = statuses;
    }
}

File: joiner-test-support/src/main/java/cz/encircled/joiner/test/model/Group.kt
package cz.encircled.joiner.test.model

import javax.persistence.*

/**
 * @author Kisel on 21.01.2016.
 */
@Entity
@Table(name = "test_group")
class Group : AbstractEntity() {

    @ManyToMany(fetch = FetchType.LAZY, mappedBy = "groups")
    var users: Set<User>? = null

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "group")
    var statuses: Set<Status>? = null
}
